### ISSUE TITLE
Why was group-based middleware support removed?

### ISSUE THREAD

**Issue by kirkbushell** [fruitcake/laravel-cors#517]
I currently have a use-case where I have an api.domain not /api setup. And I do not want to have to open up my entire application just to support the api requests.

Is there a way to support this currently, perhaps by filtering by domain somehow? If not, I'm really curious why group-based middleware support was removed =\ It would have handled my use-case perfectly.

**Comment by ftrudeau-pelcro** [comment]
I concur.

We have multiple different sites poking our API endpoints. A series of middleware must run in sequence (configured as a group of middlewares), that detects the site, amongst other things. Only after some middlewares run that HandleCors should run, as it requires context to, for instance, dynamically set the `allowed_origins`.

What is the alternative, if any ?

**Comment by ftrudeau-pelcro** [comment]
Hey @barryvdh, any plan to reinstate support for group-based middleware any time soon ? Thanks for the package btw ;)

**Comment by ftrudeau-pelcro** [comment]
Would appreciate feedback @barryvdh

**Comment by barryvdh** [comment]
I think it would work in a group, but you want different options?

**Comment by kirkbushell** [comment]
> I think it would work in a group, but you want different options?

What kind of options? This used to be part of group middleware, which allows you to group routes together nicely for CORS support, which we can no longer do.

**Comment by ftrudeau-pelcro** [comment]
> What kind of options? This used to be part of group middleware, which allows you to group routes together nicely for CORS support, which we can no longer do.

I concur.

**Comment by ftrudeau-pelcro** [comment]
> What kind of error do you het when using it as a group? If you just add to a group and set the path to allow *?

Just quoting the [documentation of latest release](https://github.com/fruitcake/laravel-cors#upgrading-from-0x--barryvdh-laravel-cors):

**Group middleware is no longer supported, use the global middleware**

**Comment by barryvdh** [comment]
Yeah it's not officially supported/tested, but I think it should work. Except error handling mostly.

**Comment by barryvdh** [comment]
I think it was a matter of adding the middleware to the group instead of the base middleware, but will check.
I had a test, but will see if I can skip some tests so we can make the rest work; https://github.com/fruitcake/laravel-cors/pull/395

**Comment by ftrudeau-pelcro** [comment]
Any progress here @barryvdh ?

**Comment by hbouhadji** [comment]
I had the same issue, and according to https://github.com/fruitcake/laravel-cors/pull/514 you can do something like

```
'paths' => [
    'api.domain' => ['*']
]
```
in the `config/cors.php`.

tested and working for me

**Comment by kirkbushell** [comment]
It definitely doesn't work at the group level. @barryvdh would be good to get some clarification on this.

**Comment by hbouhadji** [comment]
yeah you still need to put it in global but why do you need to apply it on a group if you can filter by domain with the "hack" ?
